client/components/domains/wpcom-domain-search/use-wpcom-shopping-cart-for-domain-search.ts

```diff
@@ -65,16 +66,28 @@ export const useWPCOMShoppingCartForDomainSearch = ( {
 
 	return useMemo( () => {
 		const domainItems = flowAllowsMultipleDomainsInCart ? getDomainsInCart( responseCart ) : [];
+		const isPlanInCart =
+			responseCart.products.find( ( product ) => isPlan( product ) ) !== undefined;
```

Comment:
The plan needs to be an annual one as well for the free domain, right?

Will `isPlanInCart` be true for monthly plans?